packages/driver/src/cy/commands/navigation.ts

```diff
@@ -50,27 +50,58 @@ const isValidVisitMethod = (method) => {
 const timedOutWaitingForPageLoad = (ms, log) => {
   debug('timedOutWaitingForPageLoad')
 
-  $errUtils.throwErrByPath('navigation.timed_out', {
-    args: {
-      configFile: Cypress.config('configFile'),
-      ms,
-    },
-    onFail: log,
-  })
+  const anticipatedCrossOriginHref = cy.state('anticipatingCrossOriginResponse')?.href
+
+  // Were we anticipating a cross origin page when we timed out?
+  if (anticipatedCrossOriginHref) {
+    // We remain in an anticipating state until either a load even happens or a timeout.
+    cy.isAnticipatingCrossOriginResponseFor(undefined)
+
+    // By default origins is just this location.
+    let originPolicies = [$Location.create(location.href).originPolicy]
+
+    const currentCommand = cy.queue.state('current')
+
+    if (currentCommand?.get('name') === 'origin') {
+      // If the current command is a cy.origin command, we should have gotten a request on the origin it expects.
+      originPolicies = [cy.state('latestActiveOriginPolicy')]
+    } else if (Cypress.isCrossOriginSpecBridge && cy.queue.isOnLastCommand()) {
+      // If this is a cross origin spec bridge and we're on the last command, we should have gotten a request on the origin of one of the parents.
+      originPolicies = cy.state('parentOriginPolicies')
+    }
+
+    $errUtils.throwErrByPath('navigation.cross_origin_load_timed_out', {
+      args: {
+        configFile: Cypress.config('configFile'),
+        ms,
+        crossOriginUrl: $Location.create(anticipatedCrossOriginHref),
+        originPolicies,
+      },
+      onFail: log,
+    })
+  } else {
```

Comment:
this else isn't really needed

Reply:
I flipped things around to reduce nesting, which also got rid of the else: https://github.com/cypress-io/cypress/pull/21137/commits/985f5ebc419e5bb563aeb9fd066e4bce009f6315